lib/rototiller/utilities/command.rb

```diff
@@ -5,22 +5,62 @@ class Command
     include ColorText
 
     # @return [String] the command to be used, could be considered a default
-    attr_reader :name
+    attr_accessor :name
 
     # @return [EnvVar] the ENV that is equal to this command
-    attr_reader :override_env
+    attr_accessor :override_env
+
+    # @return [String, nil] the value that should be used as an argument to the given command
+    attr_accessor :argument
+
+    # @return [EnvVar] the ENV that is equal to the argument to be used with this command
+    attr_accessor :argument_override_env
 
     # Creates a new instance of CommandFlag, holds information about desired state of a command
     # @param [Hash] attribute_hash hashes of information about the command
     # @option attribute_hash [String] :command The command
     # @option attribute_hash [String] :override_env The environment variable that can override this command
-    def initialize(attribute_hash)
-      # translate the keys from 'Command' to Env
-      if attribute_hash[:override_env]
-        @override_env = EnvVar.new({:name => attribute_hash [:override_env], :default => attribute_hash[:name]})
-        @name = @override_env.value
+    def initialize(h = {})
+      # check if an override_env is provided
+      @override_env = EnvVar.new({:name => h[:override_env], :default => h[:name]}) if h[:override_env]
+      @name = h[:name]
+
+      # check if an argument_override_env is provided
+      @argument_override_env = EnvVar.new({:name => h[:argument_override_env], :default => h[:argument]}) if h[:argument_override_env]
+      @argument = h[:argument]
+    end
+
+    def name=(name)
+      @override_env.default = name if override_env
+      @name = name
+    end
+
+    def argument=(arg)
+      argument_override_env.var = arg if argument_override_env
+      @argument = arg
+    end
+
+    def override_env=(env)
+      override_env ? override_env.var = env : @override_env = EnvVar.new({:name => env})
+    end
+
+    def argument_override_env=(env)
+      argument_override_env ? argument_override_env.var = env : @argument_override_env = EnvVar.new({:name => env})
+    end
+
+    def value
+
+      # determine the value of the command
+      if override_env && argument_override_env
+        "#{override_env.value} #{argument_override_env.value}"
+      elsif override_env
+        override_env.value
+      elsif argument_override_env
+        "#{name} #{argument_override_env.value}"
+      elsif argument
+        "#{name} #{argument}"
```

Comment:
does this imply that flags will be after args?  this probably won't work for BSDs (osx)